samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -234,6 +234,12 @@ void assignResourceRequests() {
         if (isRequestExpired(request)) {
           updateExpiryMetrics(request);
           containerManager.handleExpiredRequest(processorId, preferredHost, request, this, resourceRequestState);
+          // SAMZA-2601: to prevent infinite looping and logs filling up the disk, when host affinity is disabled,
+          // we explicitly break the loop here and the whole process gets retried in run() after allocatorSleepIntervalMs
+          if (!hostAffinityEnabled) {
+            LOG.info("Host affinity is disabled on expired request.");
```

Comment:
Replace it with something like this:

LOG.info("Waiting for resources to get allocated for request {}, no retries will be issued since host affinity is disabled", request);